extensions/wikia/CommunityPage/models/CommunityPageSpecialUsersModel.class.php

```diff
@@ -242,52 +246,98 @@ function () use ( $limit ) {
 		return $data;
 	}
 
+	private function getAllContributorsWithCurrentUser( $allContributorsData, $currentUserId ) {
+		global $wgCityId;
+
+		// Get current user's stats
+		$checkUserCallback = function( $user ) { return true; };
+		$userInfo = $this->wikiService->getUserInfo(
+			$currentUserId,
+			$wgCityId,
+			AvatarService::AVATAR_SIZE_SMALL_PLUS,
+			$checkUserCallback
+		);
+
+		if ($userInfo['lastRevision'] !== null) {
+			// Remove from all users list if present
+			// Done manually to make sure current user data is never out of sync due to cache
+			foreach ($allContributorsData as $key => $item) {
+				if ( $item['userId'] === $currentUserId) {
+					unset( $allContributorsData[$key] );
+					break;
+				}
+			}
+
+			// Add current user on top of list
+			$avatar = AvatarService::renderAvatar( $userInfo['name'], AvatarService::AVATAR_SIZE_SMALL_PLUS );
+
+			$data = [
+				'userId' => $currentUserId,
+				'latestRevision' => $userInfo['lastRevision'],
+				'timeAgo' => wfTimeFormatAgo( $userInfo['lastRevision'] ),
+				'userName' => $userInfo['name'],
+				'isAdmin' => $this->isAdmin( $currentUserId, $this->getAdmins() ),
+				'isCurrent' => true,
+				'avatar' => $avatar,
+				'profilePage' => $userInfo['userPageUrl'],
+			];
+
+			array_unshift( $allContributorsData, $data );
+		}
+
+		return $allContributorsData;
+	}
+
 	/**
 	 * Gets a list of all members of the community.
 	 * Any user who has made an edit in the last 2 years is a member
 	 *
-	 * @return array
+	 * @param null $currentUserId
+	 * @return Mixed|null
 	 */
-	public function getAllMembers() {
-		$data = WikiaDataAccess::cache(
+	public function getAllContributors( $currentUserId = null ) {
+		$allContributorsData = WikiaDataAccess::cache(
 			wfMemcKey( self::ALL_MEMBERS_MCACHE_KEY ),
-			WikiaResponse::CACHE_STANDARD,
-			function () {
+			WikiaResponse::CACHE_SHORT,
+			function () use ( $currentUserId ) {
 				$db = wfGetDB( DB_SLAVE );
 
-				$sqlData = ( new WikiaSQL() )
-					->SELECT( '*' )
-					->FROM ( 'wikia_user_properties' )
-					->LEFT_JOIN( 'user_groups' )->ON( 'wup_user = ug_user' )
-					->WHERE ( 'wup_property' )->EQUAL_TO( 'firstContributionTimestamp' )
-					->AND_ ( 'wup_value > DATE_SUB(now(), INTERVAL 2 YEAR)' )
-					->ORDER_BY( 'wup_value DESC' )
-					->runLoop( $db, function ( &$sqlData, $row ) {
-						$user = User::newFromId( $row->wup_user );
-						$userName = $user->getName();
+				$userSqlData = ( new WikiaSQL() )
+					->SELECT( 'rev_user, rev_timestamp' )
+					->FROM( 'revision' )
+					->WHERE( 'rev_timestamp > DATE_SUB(now(), INTERVAL 2 YEAR)' )
+					->AND_( 'rev_user' )->NOT_EQUAL_TO( 0 )
+					->GROUP_BY( 'rev_user' )
+					->ORDER_BY( 'rev_timestamp DESC' )
+					->LIMIT( self::ALL_CONTRIBUTORS_MODAL_LIMIT )
+					->runLoop( $db, function ( &$userSqlData, $row ) use ( $currentUserId ) {
```

Comment:
`$currentUserId` isn't used here, but you're adding the current user afterwards regardless. Can we only call `getAllContributorsWithCurrentUser` if the current user isn't added here rather than looping over it again to remove them if they are?